app/javascript/react/pages/ApplicantsUpload.jsx

```diff
@@ -141,13 +141,23 @@ export default function ApplicantsUpload({ organisation, configuration, departme
   };
 
   const retrieveUpToDateApplicants = async (applicantsFromList) => {
+    const departmentInternalIds = applicantsFromList
+      .map((applicant) => applicant.departmentInternalId)
+      .filter((departmentInternalId) => departmentInternalId);
     const uids = applicantsFromList.map((applicant) => applicant.uid).filter((uid) => uid);
-    let upToDateApplicants = applicantsFromList;
 
-    const retrievedApplicants = await retrieveApplicantsFromApp(uids);
+    const retrievedApplicants = await retrieveApplicantsFromApp(departmentInternalIds, uids);
 
-    upToDateApplicants = applicantsFromList.map((applicant) => {
-      const upToDateApplicant = retrievedApplicants.find((a) => a.uid === applicant.uid);
```

Comment:
Pourquoi ne pas garder le code présent et itérer qu'une fois sur les `retrievedApplicants` plutôt que 2. On ferait le check sur le `departmentInternald` et l'`uid` en même temps:  
```javascript
const upToDateApplicant = retrievedApplicants.find((a) =>(a.departmentInternalId && a.departmentInternalId === applicant.departmentInternalId) || (a.uid && a.uid === applicant.uid));
```